test/helpers/foundry/download.ts

```diff
@@ -0,0 +1,82 @@
+import { request as httpRequest, type IncomingMessage } from 'node:http';
+import { request as httpsRequest } from 'node:https';
+import { Stream } from 'node:stream';
+import { pipeline } from 'node:stream/promises';
+import { DownloadOptions } from './types';
+
+class DownloadStream extends Stream.PassThrough {
+  async response(): Promise<IncomingMessage> {
+    return new Promise((resolve, reject) => {
+      this.once('response', resolve);
+      this.once('error', reject);
+    });
+  }
```

Comment:
In a `HEAD` only request, there is no `body`, so there is no data to send to the stream. This `response` function is used to return the headers (and `status`, `statusCode`, etc). outside of the stream itself